apps/web-mzima-client/src/app/feed/feed.component.ts

```diff
@@ -725,9 +730,15 @@ export class FeedComponent extends MainViewComponent implements OnInit, OnDestro
 
   public changePostsStatus(status: string): void {
     if (status === PostStatus.Published) {
-      const uncompletedPosts: PostResult[] = this.selectedPosts.filter(
-        (post: PostResult) => !postHelpers.isAllRequiredCompleted(post),
-      );
+      const uncompletedPosts: PostResult[] = this.selectedPosts.filter((post: PostResult) => {
+        if (post.post_content) {
+          !postHelpers.isAllRequiredCompleted(post);
+        } else {
+          this.postsService.getById(post.id).subscribe((fetchedPost: PostResult) => {
+            !postHelpers.isAllRequiredCompleted(fetchedPost);
+          });
+        }
```

Comment:
Just a tiny thing, you need to return the result of the isAllRequiredCompleted to be ably to save it in the uncompletedPosts array 👍 If you do that, it works fine!